File: Source/TS/game/map.ts
import { Grid } from "../utilities/Grid";
import { Tile } from "./Tile";

export class Map extends Grid<Tile>
{
    constructor(width: number, height: number)
    {
        super(width, height);

        for (let y = 0; y < this.dimension.y; y++)
        {
            for (let x= 0; x < this.dimension.x; x++)
            {
                // TODO: map out tiles
            }
        }
    }

    set(x: number, y: number, value: Tile): void 
    {
        throw new Error("Method not implemented.");
    }

    get(x: number, y: number): Tile | null 
    {
        throw new Error("Method not implemented.");
    }
}

File: core/src/main/java/io/github/RimIsland/Game/Map.java
package io.github.RimIsland.Game;
import io.github.RimIsland.Utilities.Grid;

public class Map extends Grid<Tile>
{
    protected Map(int cols, int rows) {
        super(cols, rows);
    }

    @Override
    public void set(int col, int row, Tile object) {
        System.out.println("Not Implemented");
    }

    @Override
    public Tile get(int col, int row) {
        System.out.println("Not Implemented");
        return null;
    }
}
